src/widget/button.rs

```diff
@@ -47,34 +45,39 @@ impl Label {
         ctx.add(self, &[])
     }
 
-    fn get_layout(&self, dwrite_factory: &directwrite::Factory) -> TextLayout {
+    fn get_layout(
+        &self,
+        rt: &mut Piet,
+        font_size: f32,
+    ) -> <<Piet as RenderContext>::Text as Text>::TextLayout {
```

Comment:
This is not super pleasant to have to write out. Since TextLayout is also the name of a trait, I'm not super sure what the best way to expose this from `piet_common` would be.